src/app/project/project-documents/detail/detail.component.html

```diff
@@ -61,5 +61,10 @@ <h3 for="documentFileName">Filename: </h3>
       <h3 for="internalSize">Filesize: </h3>
       <p id="internalSize">{{ (document && document.internalSize) ? humanReadableSize : '0' }}</p>
     </div>
+    <div class="label-pair">
+      <h3 for="url">URL: </h3>
+      <a *ngIf="'Publish' !== publishText" id="url" target="blank" href="{{documentUrl || '-'}}">{{documentUrl || '-'}}</a>
+      <span *ngIf="'Publish' === publishText">-</span>
```

Comment:
Am I reading this right? If *not* published, show the file URL, and if published, show "-"?

Reply:
So as with other places in the file, this checks for the state of the "publish" or "un-publish" button. The logic being that if you're currently able to publish, then the file must be un-published at that time. In retrospect it is confusing to read. I'll update it to use a more readable way to toggle the URL showing.